### ISSUE TITLE
[a11y]: Tooltip examples and behaviour need to be updated

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#12921]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

https://react.carbondesignsystem.com/?path=/story/components-tooltip--default

### Automated testing tool and ruleset

n/a

### Description

The interaction of various tip examples is inconsistent and contradictory between tooltip and toggletip..
~1) The Live demo example on the [usage page](https://carbondesignsystem.com/components/tooltip/usage#live-demo) makes no sense. When the focus lands on the i icon, nothing happens. Then I tab again, and focus remains on the 'i' and suddenly it has a tip. Focus should immediately trigger thte tip to appear.~
~2) In the same Live demo section, changing the Variant selector to be definition tooltip also shows an implementation problem. The definition never appears on focus/hover but instead appears exclusively on click/activate. It is implemented as a TOGGLETIP instead of as a tooltip.~
3) In storybook, the definition tooltip is under its own "D" heading instead of being under tooltip. It is less likely to be discovered there, and should be under tooltip, as is the case with variants of other components
4) Further the definition tooltip is again implemented as a toggletip instead of a tooltip. The difference in behaviour makes no  sense. As mentioned in https://github.com/carbon-design-system/carbon-website/issues/3332, the definition tooltip should always expose on hover/focus.
5) The [playground](https://react.carbondesignsystem.com/?path=/story/components-tooltip--playground) for tooltip has the tooltip appearing before it has focus. Same for the [playground for the definition tooltip](https://react.carbondesignsystem.com/?path=/story/components-definitiontooltip--playground). This should obviously not be happening
6) As mentioned in the website issue, all the uses of the information icon ('i') for tooltips should be expunged from the storybook and from the Figma kits
![image](https://user-images.githubusercontent.com/14298245/210643732-6e84466e-be27-4beb-8f53-6adf762c11a7.png)
7) A challenge for Carbon is that while the tooltip component has a requirement for text, the icon-only button does not, nor does the ghost button, yet they both need a tooltip to provide context.  So guidance in both Carbon and storybook and in Figman's buttons should be revamped so that the label/name attribute is populated -- except in some very limited cases where there is a reason that the tip should _not_ be exposed on focus/hover.

```[tasklist]
### Tasks
- [ ] https://github.com/carbon-design-system/carbon-website/issues/3712
- [ ] [Tooltip a11y]: Playground implementation
- [ ] https://github.com/carbon-design-system/carbon-design-kit/issues/721
```

### Reproduction/example

https://carbondesignsystem.com/components/tooltip/usage#live-demo

### Steps to reproduce

Go to both the live demo and the storybook examples
- https://carbondesignsystem.com/components/tooltip/usage#live-demo
- https://react.carbondesignsystem.com/?path=/story/components-definitiontooltip--default
- https://react.carbondesignsystem.com/?path=/story/components-tooltip--default

Using a keyboard
1. attempt to have tooltips display on focus (tab to them without having to activate)
2. attempt to dismiss using Esc

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by mbgower** [comment]
@sstrubberg IMO issues 4 and 5 should be treated as higher severity.

**Comment by sstrubberg** [comment]
@mbgower Any chance you could break issues 4 and 5 into separate tickets that we can triage and prioritize against a sprint? Also happy to have you come back to refinement so we can do it together :)

**Comment by mbgower** [comment]
> Also happy to have you come back to refinement so we can do it together :)

@sstrubberg Happy to. Can you please add me to invite

**Comment by laurenmrice** [comment]
**Addressing mentioned above items** 

> 1. The Live demo example on the [usage page](https://carbondesignsystem.com/components/tooltip/usage#live-demo) makes no sense. When the focus lands on the i icon, nothing happens. Then I tab again, and focus remains on the 'i' and suddenly it has a tip. Focus should immediately trigger thte tip to appear.
> 
> 2. In the same Live demo section, changing the Variant selector to be definition tooltip also shows an implementation problem. The definition never appears on focus/hover but instead appears exclusively on click/activate. It is implemented as a TOGGLETIP instead of as a tooltip.

- Points 1 and 2: Made an issue about the live demo problems in the carbon-website repo. [#3712](https://github.com/carbon-design-system/carbon-website/issues/3712)

----

> 3. In storybook, the definition tooltip is under its own "D" heading instead of being under tooltip. It is less likely to be discovered there, and should be under tooltip, as is the case with variants of other components.

> 4. Further the definition tooltip is again implemented as a toggletip instead of a tooltip. The difference in behavior makes no sense. As mentioned in https://github.com/carbon-design-system/carbon-website/issues/3332, the definition tooltip should always expose on hover/focus.

> 5. The [playground](https://react.carbondesignsystem.com/?path=/story/components-tooltip--playground) for tooltip has the tooltip appearing before it has focus. Same for the [playground for the definition tooltip](https://react.carbondesignsystem.com/?path=/story/components-definitiontooltip--playground). This should obviously not be happening.

> 6. As mentioned in the website issue, all the uses of the information icon ('i') for tooltips should be expunged from the storybook and from the Figma kits

- Points 3, 4, 5, and 6 are dev tasks.
  - For the fourth bullet point, definition tooltip can optionally open **on click**, this should be a prop or optional implementation. 
  - For the sixth bullet point, the figma portion of this will be tracked in an issue in the carbon-design-kit repo. [#721](https://github.com/carbon-design-system/carbon-design-kit/issues/721)

**Comment by tay1orjones** [comment]
We've updated the live demo on the website, so I crossed out the first two items in the list.

**Comment by tay1orjones** [comment]
`#3` - This will be covered by #13189

`#4` - They're always able to be invoked on click, we have a prop for disabling the hover interaction though. We do need to update DefinitionTooltip to open on focus.

`#5` - We can remove `defaultOpen` from the playground stories for all tooltips

So `#4` and `#5` are the only actionable pieces left for this issue. We can handle those together in one PR, once done we can close this issue.
